test/Microsoft.AspNet.Mvc.FunctionalTests/TagHelperSampleTest.cs

```diff
@@ -1,106 +1,55 @@
-        private readonly Action<IServiceCollection> _configureServices = new TagHelperSample.Web.Startup().ConfigureServices;
-
-        [Fact]
-        public async Task Home_Pages_ReturnSuccess()
+        public TagHelperSampleTest(MvcTestFixture<TagHelperSample.Web.Startup> fixture)
         {
-            // Arrange
-            var server = TestHelper.CreateServer(app => _app(app, _loggerFactory), SiteName, SamplesFolder, _configureServices);
-            var client = server.CreateClient();
-
-            for (var index = 0; index < Paths.Count; index++)
-            {
-                // Act
-                var path = Paths[index];
-                var response = await client.GetAsync("http://localhost" + path);
-
-                // Assert
-                Assert.NotNull(response);
-                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
-            }
+            Client = fixture.Client;
         }
 
-        private class TestLoggerFactory : ILoggerFactory
-        {
-            public LogLevel MinimumLevel { get; set; }
-
-            public void AddProvider(ILoggerProvider provider)
-            {
-
-            }
+        public HttpClient Client { get; }
 
-            public ILogger CreateLogger(string name)
-            {
-                return new TestLogger();
-            }
-
-            public void Dispose()
-            {
-            }
-        }
-
-        private class TestLogger : ILogger
+        public static TheoryData<string> PathData
         {
-            public bool IsEnabled(LogLevel level)
-            {
-                return false;
-            }
-
-            public IDisposable BeginScopeImpl(object scope)
-            {
-                return new TestDisposable();
-            }
-
-            public void Log(LogLevel logLevel, int eventId, object state, Exception exception, Func<object, Exception, string> formatter)
-            {
+            get
+            {
+                return new TheoryData<string>
+                {
+                    string.Empty,
+                    "/",
+                    "/Home/Create",
+                    "/Home/Create?Name=Billy&Blurb=hello&DateOfBirth=2000-11-30&YearsEmployeed=0",
+                    "/Home/Create",
+                    "/Home/Create?Name=Joe&Blurb=goodbye&DateOfBirth=1980-10-20&YearsEmployeed=1",
+                    "/Home/Edit/0",
+                    "/Home/Edit/0?Name=Bobby&Blurb=howdy&DateOfBirth=1999-11-30&YearsEmployeed=1",
+                    "/Home/Edit/1",
+                    "/Home/Edit/1?Name=Jack&Blurb=goodbye&DateOfBirth=1979-10-20&YearsEmployeed=4",
+                    "/Home/Edit/0",
+                    "/Home/Edit/0?Name=Bobby&Blurb=howdy&DateOfBirth=1999-11-30&YearsEmployeed=2",
+                    "/Home/Index",
+                };
             }
         }
 
-        private class TestDisposable : IDisposable
+        [Theory]
+        [MemberData(nameof(PathData))]
```

Comment:
Two overlapping changes aren't easy to read here:
- don't need `TestLoggerFactory` plus `NullLoggerFactory`
- `Home_Pages_ReturnSuccess()` should never have been a single test with a loop in it